### PR TITLE
libs: Handle SIGHUP explicitly inside autofile

### PR DESCRIPTION
**Pull request by melekes**
Refs #2260

* [ ] ~~Updated all relevant documentation in docs~~
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [ ] ~~Updated CHANGELOG_PENDING.md~~

### CODE REVIEW

File: libs/autofile/autofile.go [hunk 1]
```diff
@@ -32,50 +34,70 @@ if err != nil {
...
-	ticker        *time.Ticker
-	tickerStopped chan struct{} // closed when ticker is stopped
-	mtx           sync.Mutex
-	file          *os.File
+	ID   string
+	Path string
+
+	closeTicker       *time.Ticker
+	closeTickerStopCh chan struct{} // closed when closeTicker is stopped
+	hup               chan os.Signal
```
Comment (melekes) [inline libs/autofile/autofile.go]: `hupc` or `hupCh`?

File: libs/autofile/autofile_test.go [hunk 2]
```diff
@@ -87,7 +83,4 @@ func TestOpenAutoFilePerms(t *testing.T) {
 	} else {
 		t.Errorf("unexpected error %v", e)
 	}
-
-	err = af.Close()
```
Comment (melekes) [inline libs/autofile/autofile_test.go]: note to reviewers: OpenAutoGroup now returns nil if there's an error

File: libs/autofile/autofile.go (hunk 1 above)
Comment (xla) [reply libs/autofile/autofile.go]: `ch` and `Ch` have a very distinct meaning in our codebase as they represent the reactor communication channels. So I'd lean towards `hupc` and try to use it consistently in the codebase. Attempted that with the recent Transport addition in `p2p`.

File: libs/autofile/autofile.go [hunk 3]
```diff
@@ -32,50 +34,70 @@ if err != nil {
...
-	Path          string
-	ticker        *time.Ticker
-	tickerStopped chan struct{} // closed when ticker is stopped
-	mtx           sync.Mutex
-	file          *os.File
+	ID   string
+	Path string
+
+	closeTicker       *time.Ticker
+	closeTickerStopCh chan struct{} // closed when closeTicker is stopped
```
Comment (xla) [inline libs/autofile/autofile.go]: Probably best to rename this one as well.

**Pull request merged**